uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+          });
+
+          it('returns badge-danger', function () {
+            expect(returnValue).toBe('badge-danger');
+          })
+        });
+
+        describe('when status is cancelled', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('cancelled')
+          });
+
+          it('returns blank string', function () {
+            expect(returnValue).toBe('');
+          })
+        });
+
+        describe('when status is all', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('all')
+          });
+
+          it('returns blank string', function () {
+            expect(returnValue).toBe('');
+          })
+        });
+
+        describe('when status is some other value', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge(jasmine.any(String))
+          });
+
+          it('returns blank string', function () {
+            expect(returnValue).toBe('badge-primary');
+          })
+        });
+      });
+
+      describe('refreshWithFilter', function () {
+
+        var mockStatus;
+
+        beforeEach(function () {
+          spyOn(controller, 'refresh');
+          mockStatus = optionGroupMock.getCollection('hrleaveandabsences_leave_request_status')[0];
+          controller.refreshWithFilter(mockStatus);
+        });
+
+        it('refreshes the data', function () {
+          expect(controller.refresh).toHaveBeenCalled();
+        });
+
+        it('sets the leaveStatus', function () {
+          expect(controller.filters.leaveRequest.leaveStatus).toEqual(mockStatus);
+        });
+      });
+
+      describe('getUserNameByID', function () {
+
+        var returnValue;
+
+        beforeEach(function () {
+          returnValue = controller.getUserNameByID(controller.filteredUsers[0].id);
+        });
+
+        it('returns name of the user', function () {
+          expect(returnValue).toBe(controller.filteredUsers[0].display_name);
+        });
+      });
+
+      describe('refresh::', function () {
+
+        describe('page no::', function () {
+
+          describe('when no page no is sent', function () {
+
+            beforeEach(function () {
+              controller.refresh();
+            });
+
+            it('page no is set to 1', function () {
+              expect(controller.pagination.page).toBe(1);
+            });
+          });
+
+          describe('when page no is sent', function () {
+
+            var pageNo = 5;
+            beforeEach(function () {
+              controller.refresh(pageNo);
+            });
+
+            it('page no is set to 1', function () {
+              expect(controller.pagination.page).toBe(pageNo);
+            });
+          });
+        });
+
+        describe('content loading', function () {
+
+          beforeEach(function () {
+            controller.refresh();
+          });
+
+          it('is shown', function () {
```

Comment:
title is weird, "is shown" what? alternative would be "it is in loading mode", or "shows spinners", or anything like that